pkg/asset/machines/azure/capabilities.go

```diff
@@ -0,0 +1,41 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/installer/pkg/types/azure"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func getHyperVGenerationVersion(capabilities map[string]string) (string, error) {
+	if val, ok := capabilities["HyperVGenerations"]; ok {
+		generations := sets.NewString()
+		for _, g := range strings.Split(val, ",") {
+			g = strings.TrimSpace(g)
+			g = strings.ToUpper(g)
+			generations.Insert(g)
+		}
+		if generations.Has("V2") {
+			return "V2", nil
+		}
+		return "V1", nil
+	}
+	return "", fmt.Errorf("unable to determine HyperVGeneration version")
+}
+
+func getVMNetworkingCapability(capabilities map[string]string) (bool, error) {
+	val, ok := capabilities[string(azure.AcceleratedNetworkingEnabled)]
+	if !ok {
```

Comment:
My understanding is that if the capability was not included, that is the equivalent of false or not supoorted--not an error. WDYT @r4f4? I think that should be the case here and in `getUltraSSDCapability`

Reply:
Yeah, that's the case for the non-zonal case of the UltraSSD. For the networking, the docs [1] say:

> "Accelerated networking is available in all global Azure regions and Azure Government Cloud"

So my understanding is that the only case when the capability will be missing is for Azure Stack Cloud. In that case, I think it's safe to say the capability is not supported.

[1] https://docs.microsoft.com/en-us/azure/virtual-network/accelerated-networking-overview#regions

Reply:
The full ultraSSD validation gets a bit more complicated in the multi-zones case because the capability is not listed with the other instance capabilites but in the `LocationInfo`. So the absence of the key from the map is not enough to determine lack of support.

Reply:
I see. I think the solution to this is to have `GetVMCapabilities` return the `typeMeta` (or perhaps just the `LocationInfo`) in addition to the more user-friendly capabilities map. We can handle that in #5854 and does not need to be a part of this PR.